### PR TITLE
[TOOLS-471] refactoring to replace jira rest client in issue searches…

### PR DESCRIPTION
**Pull request by taksan**
… and retrieval

### CODE REVIEW

File: src/main/java/objective/taskboard/jira/JiraService.java [hunk 1]
```diff
@@ -174,10 +175,10 @@ public String getResolutions(String transitionName) {
         return ImmutableList.copyOf(response);
     }
 
-    public Optional<Issue> getIssueByKey(String key) {
+    public Optional<JiraIssueDto> getIssueByKey(String key) {
         log.debug("⬣⬣⬣⬣⬣  getIssueByKey");
         try {
-            return Optional.of(jiraEndpointAsUser.executeRequest(client -> client.getIssueClient().getIssue(key)));
+            return Optional.of(jiraEndpointAsUser.request(JiraIssueDto.Service.class).get(key));//(client -> client.getIssueClient().getIssue(key)));
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/jira/JiraService.java]: BLOCKER This block of commented-out lines of code should be removed. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![BLOCKER][BLOCKER] 1 blocker

Watch the comments in this conversation to review them.

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'

File: src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java [hunk 2]
```diff
@@ -303,7 +302,11 @@ public static String extractReleaseId(JiraProperties jiraProperties, Issue issue
         return result;
     }
 
-    private static void logErrorExtractField(Issue issue, IssueField field, JSONException e) {
+    private static void logErrorExtractField(JiraIssueDto issue, JiraIssueFieldDto field, JSONException e) {
         log.error("Error extracting " + field.getName() + " from issue " + issue.getKey() + ": " + e.getMessage());
-    }    
+    }
+
+    private static boolean isEmpty(List<JiraLinkDto> issueLinks) {
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/domain/converter/IssueFieldsExtractor.java]: import static org.springframework.util.CollectionUtils.isEmpty;

File: src/main/java/objective/taskboard/jira/client/JiraCommentDto.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package objective.taskboard.jira.client;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+@JsonIgnoreProperties(ignoreUnknown=true)
+public class JiraCommentDto {
+
+}
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/jira/client/JiraCommentDto.java]: Já não era usado antes, não seria interessante remover de vez? Ou deixa pra próxima iteração de refactor?

File: src/main/java/objective/taskboard/jira/client/JiraIssueDto.java [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+    }
+
+    public JiraIssueFieldDto getField(String field) {
+        JSONObjectAdapter jsonObjectAdapter = fields.other.get(field);
+        if (jsonObjectAdapter == null)
+            return null;
+        return new JiraIssueFieldDto(field, jsonObjectAdapter.object);
+    }
+    
+    public Map<String, JSONObjectAdapter> fields() {
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/jira/client/JiraIssueDto.java]: Este método é necessário?

File: src/main/java/objective/taskboard/jira/client/JiraIssueTypeDto.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    public URI getIconUri() {
+        return iconUrl;
+    }
+    
+    public String getName() {
+        return name;
+    }
+
+    public boolean isSubtask() {
+        throw new RuntimeException("NOT IMPLEMENTED");
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/jira/client/JiraIssueTypeDto.java]: Não é usado de fato. Não seria o caso de remover esta property?

File: src/main/java/objective/taskboard/jira/data/Version.java [hunk 6]
```diff
@@ -13,9 +14,8 @@
     public final String id;
     public final String name;
 
-    public Version(
-            @JsonProperty("id") String id
-            , @JsonProperty("name") String name) {
+    @JsonCreator
+    public Version(@JsonProperty("id") String id, @JsonProperty("name") String name) {
```
Comment (herbert-venancio) [inline src/main/java/objective/taskboard/jira/data/Version.java]: Este não segue o padrão

File: src/main/java/objective/taskboard/jira/client/JiraTimeTrackingDto.java [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+public class JiraTimeTrackingDto {
+    
+    @JsonProperty
+    public Integer originalEstimateSeconds;
+    @JsonProperty
+    public Integer timeSpentSeconds;
+    
+    public JiraTimeTrackingDto(){}
+
+    public JiraTimeTrackingDto(Integer timeEstimateMinutes, Object object, Integer timeSpentMinutes) {
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/jira/client/JiraTimeTrackingDto.java]: MAJOR Remove this unused method parameter "object". [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS1172)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**